src/format.rs

```diff
@@ -214,7 +235,10 @@ impl Printf for i8 {
 
 impl Printf for u8 {
     fn format(&self, spec: &ConversionSpecifier) -> Result<String> {
-        (*self as u64).format(spec)
+        match spec.conversion_type {
+            ConversionType::Char => char::from(*self).format(spec),
```

Comment:
I'm not sure about this - I think this will misinterpret bytes from multi-byte UTF-8 characters

Reply:
I see, then restricting u8 to ASCII should be fine.